depends/packages/bdb.mk

```
@@ -32,5 +32,12 @@ define $(package)_build_cmds
 endef
 
 define $(package)_stage_cmds
-  $(MAKE) DESTDIR=$($(package)_staging_dir) install_lib install_include
+  $(MAKE) DESTDIR=$($(package)_staging_dir) install
+endef
+
+define $(package)_postprocess_cmds
+  cd $($(package)_staging_dir) && \
+  cd ../../../../../../zcutil && \
+  mkdir -p bdb && \
+  mv -f $($(package)_staging_dir)$(host_prefix)/bin/db_* bdb
```

Make this folder tool-agnostic, so we can use it for any helper tools that are generated during the build process:
```suggestion
  mkdir -p bin && \
  mv -f $($(package)_staging_dir)$(host_prefix)/bin/db_* bin
```
Also remove the folder in the `clean-local` step of the root `Makefile`.

---

I made the suggested changes and force pushed however unsure about the clean.

With this change, when i install from scratch the `zcutil/bin` directory will be created, but then if i do `make clean` and build again(with `./zcutil/build.sh -j$(nproc)`) the folder will not be created again. Not sure if this is expected behaviour.

---

Ah, this would indeed be the expected behaviour, due to the way that the depends system works. The `postprocess` step of a dependency build is for cleaning up dependency files after staging. As the `bdb` dependency doesn't require rebuilding, the `postprocess` step is not re-run; the BDB binaries are simply extracted from the already-postprocessed tarball.

We definitely want `make clean` to remove `zcutil/bin` as it contains entirely-generated files. I think instead the `postprocess` step should be moved into the main `Makefile.am`; that way, the `zcutil/bin` directory will be re-created (and the tools re-copied from staging) after a `make clean`. We could do this here, or in a follow-up PR.